### PR TITLE
Added BLAS level2 function GEMV to the linear algebra benchmarks

### PR DESCRIPTION
**Pull request by nadir199**
Hi,

Please find my implementation of GEMV level2 BLAS function. The code performs 
       out = alpha * A * x + beta * y
Where 
     **A** is a M x N matrix
     **x** is a size N vector
     **y** is a size M vector
    **alpha**, **beta** are scalars

Thank you.

### CODE REVIEW

File: benchmarks/linear_algebra/blas/level2/gemv/gemv_generator.cpp [hunk 1]
```diff
@@ -0,0 +1,129 @@
+#include <tiramisu/tiramisu.h>
+
+using namespace tiramisu;
+
+
+/**
+ * Benchmark for BLAS GEMV
+ *     out = a*A*x + b*y
+ *  
+ *
```
Comment (rbaghdadi) [inline benchmarks/linear_algebra/blas/level2/gemv/gemv_generator.cpp]: Can you please remove extra new lines? Keep only one new line.

File: benchmarks/linear_algebra/blas/level2/gemv/gemv_generator.cpp [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+{
+    tiramisu::global::set_default_tiramisu_options();
+
+    // -------------------------------------------------------
+    // Layer I
+    // -------------------------------------------------------
+
+    tiramisu::function func(name);
+
+
```
Comment (rbaghdadi) [inline benchmarks/linear_algebra/blas/level2/gemv/gemv_generator.cpp]: No need for an extra new line here for example.

File: benchmarks/linear_algebra/blas/level2/gemv/gemv_generator.cpp [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+    // -------------------------------------------------------
+
+    tiramisu::function func(name);
+
+    
+    //Declare both iteration variables
+    tiramisu::var i("i");
+    tiramisu::var j("j");
+	//Matrix Dimensions
+    tiramisu::computation SIZES("{SIZES[k]: 0<=k<2}", tiramisu::expr(), false, p_float64, &func);
```
Comment (rbaghdadi) [inline benchmarks/linear_algebra/blas/level2/gemv/gemv_generator.cpp]: Can you use the new syntax of Tiramisu instead of the old one?

Comment (rbaghdadi) [reply benchmarks/linear_algebra/blas/level2/gemv/gemv_generator.cpp]: You find the new syntax in the first two or three tutorials.

File: benchmarks/linear_algebra/blas/level2/gemv/gemv_generator.cpp [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+    tiramisu::buffer buf_A("buf_A", {M,N}, tiramisu::p_float64, a_input, &func);
+    tiramisu::buffer buf_x("buf_x", {N}, tiramisu::p_float64, a_input, &func);
+    tiramisu::buffer buf_y("buf_y", {M}, tiramisu::p_float64, a_input, &func);
+    tiramisu::buffer buf_alpha("buf_alpha", {1}, tiramisu::p_float64, a_input, &func);
+    tiramisu::buffer buf_beta("buf_beta", {1}, tiramisu::p_float64, a_input, &func);
+
+    //OUTPUT BUFFERS
+    tiramisu::buffer buf_result("buf_result",{M}, tiramisu::p_float64,a_output,&func);
+
+	SIZES.set_access("{SIZES[k]->buf_SIZES[k] : 0<=k<2}");
```
Comment (rbaghdadi) [inline benchmarks/linear_algebra/blas/level2/gemv/gemv_generator.cpp]: A problem in indentation here.

File: benchmarks/linear_algebra/blas/level2/gemv/gemv_generator.cpp [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+    // -------------------------------------------------------
+    // Code Generation
+    // -------------------------------------------------------
+
+    func.set_arguments({&buf_SIZES,&buf_A, &buf_x, &buf_y,&buf_alpha,&buf_beta,&buf_result});
+    func.gen_time_space_domain();
+    func.gen_isl_ast();
+    func.gen_halide_stmt();
+    func.gen_halide_obj("generated_" + name + ".o");
+}
```
Comment (rbaghdadi) [inline benchmarks/linear_algebra/blas/level2/gemv/gemv_generator.cpp]: Use the new syntax here also. Check the first two tutorials.

File: benchmarks/linear_algebra/blas/level2/gemv/gemv_generator.cpp (hunk 1 above)
Comment (nadir199) [reply benchmarks/linear_algebra/blas/level2/gemv/gemv_generator.cpp]: Sorry for that. Done.

File: benchmarks/linear_algebra/blas/level2/gemv/gemv_generator.cpp (hunk 3 above)
Comment (nadir199) [reply benchmarks/linear_algebra/blas/level2/gemv/gemv_generator.cpp]: Thank you. It's done.

**Pull request merged**